codebase/general/src.lib/dmap.1.25/include/dmap.h

```diff
@@ -151,7 +151,7 @@ void *DataMapStoreArray(struct DataMap *ptr,
 int DataMapRemoveArray(struct DataMap *ptr,char *name,int type,int dim);
 
 void *DataMapFindArray(struct DataMap *ptr,char *name,int type,int dim,
-		       int **rng);
+		       int32 **rng);
```

Comment:
What's the reason this particular variable needs be labeled int32?

Reply:
Because that's how rng is declared in DataMapFindArray in `dmap.c`:

https://github.com/SuperDARN/rst/blob/bfa8cb951ddf93126d3a244cbedccdbaac045f9f/codebase/general/src.lib/dmap.1.25/src/dmap.c#L636-L637

And earlier in `dmap.h`:

https://github.com/SuperDARN/rst/blob/bfa8cb951ddf93126d3a244cbedccdbaac045f9f/codebase/general/src.lib/dmap.1.25/include/dmap.h#L93-L100

As noted in https://github.com/vtsuperdarn/davitpy/pull/40, DataMapFindArray isn't actually used anywhere in the RST but it has been cited as causing compilation problems in other situations.

Reply:
Well, looking at the DataMapArray struct, dim is also labeled int32, so maybe that's not the best justification.  More importantly I think, all other functions in dmap.c and dmap.h label rng as int32 rather than int.

Reply:
I would agree. If you ever want to break your brain, look at rtypes.h. It actually does make sure that variables are all the same size on any platform, but I think it was written before the std types lib was available. I would love if we could get rid of that file.

Reply:
As a C dummy, what would prevent us from doing so?  time/effort?  Or is it a compatibility issue with the ROS side of things?

Reply:
I dont think it would be too hard to do in RST. You could easily have a script that would search and replace int, short, char with int32,int16,int8 or whatever. As long as we make sure that rtypes define the variables to be the same sizes as those in the std types lib, I think it would work and be compatible?

Reply:
I agree, and yes this look fine to me. There is more we need to consider for a future PR on changing types.